modules/fedimint-ln-client/src/receive.rs

```diff
@@ -0,0 +1,385 @@
+            }
+            LightningReceiveStates::ConfirmedInvoice(confirmed_invoice) => {
+                confirmed_invoice.transitions(global_context)
+            }
+            LightningReceiveStates::Funded(funded) => {
+                funded.transitions(self.common.clone(), global_context)
+            }
+            LightningReceiveStates::Success(_) => {
+                vec![]
+            }
+        }
+    }
+
+    fn operation_id(&self) -> fedimint_client::sm::OperationId {
+        self.common.operation_id
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Decodable, Encodable)]
+pub struct LightningReceiveSubmittedOffer {
+    pub offer_txid: TransactionId,
+    pub invoice: Invoice,
+    pub payment_keypair: KeyPair,
+}
+
+#[derive(Error, Clone, Debug, Serialize, Deserialize, Encodable, Decodable, Eq, PartialEq)]
+pub enum LightningReceiveError {
+    #[error("Offer transaction was rejected")]
+    Rejected,
+    #[error("Incoming Lightning invoice was not paid within the timeout")]
+    Timeout,
+    #[error("Claim transaction was rejected")]
+    ClaimRejected,
+}
+
+impl LightningReceiveSubmittedOffer {
+    fn transitions(
+        &self,
+        common: &LightningReceiveCommon,
+        global_context: &DynGlobalClientContext,
+        context: &LightningClientContext,
+    ) -> Vec<StateTransition<LightningReceiveStateMachine>> {
+        let global_context = global_context.clone();
+        let rejected_context = global_context.clone();
+        let txid = self.offer_txid;
+        let invoice = self.invoice.clone();
+        let payment_keypair = self.payment_keypair;
+        let rejected_common = common.clone();
+        vec![
+            StateTransition::new(
+                Self::await_invoice_confirmation(
+                    global_context,
+                    context.ln_decoder.clone(),
+                    txid,
+                    invoice,
+                    payment_keypair,
+                ),
+                move |_dbtx, result, old_state| {
+                    Box::pin(Self::transition_confirmed_invoice(result, old_state))
+                },
+            ),
+            StateTransition::new(
+                Self::await_invoice_rejection(rejected_context, rejected_common, txid),
+                |_dbtx, (), old_state| Box::pin(Self::transition_invoice_rejected(old_state)),
+            ),
+        ]
+    }
+
+    async fn await_invoice_confirmation(
+        global_context: DynGlobalClientContext,
+        module_decoder: Decoder,
+        txid: TransactionId,
+        invoice: Invoice,
+        payment_keypair: KeyPair,
+    ) -> Result<ConfirmedInvoice, LightningReceiveError> {
+        let outpoint = OutPoint { txid, out_idx: 0 };
+        let timeout = Duration::from_secs(15);
+        global_context
+            .api()
+            .await_output_outcome::<LightningOutputOutcome>(outpoint, timeout, &module_decoder)
+            .await
+            .map_err(|_| LightningReceiveError::Rejected)?;
```

Comment:
Since we aren't doing anything with the output itself we could just await the transaction being accepted via the `await_tx_accepted` fn on the global context. That way we don't need the decoder.

Even better, that way the `await_invoice_rejection` wouldn't be needed since `await_tx_accepted` returns a result that can be used to decide to which following state to transition.